### PR TITLE
Changed DetailsRowCheck to <button>, from <div>, so Narrator announces state changes when user toggles the checkbox

### PR DESCRIPTION
**Pull request by HalcyonPride**
#### Pull request checklist

- [X] Addresses an existing issue: https://onedrive.visualstudio.com/OneDriveWeb/_workitems/edit/517318
- [X] Include a change request file using `$ npm run change`

#### Description of changes

As described

#### Focus areas to test

DetailsList

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx [hunk 1]
```diff
@@ -48,10 +48,11 @@ export const DetailsRowCheck = (props: IDetailsRowCheckProps) => {
       aria-checked={ isPressed }
       data-selection-toggle={ true }
       data-automationid='DetailsRowCheck'
+      type='button'
```
Comment (ThomasMichon) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx]: Declaring a `<button>` as `type='button'` is redundant.

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ export const DetailsRowCheck = (props: IDetailsRowCheckProps) => {
   const isPressed = props.isSelected || props.selected;
 
   return (
-    <div
+    <button
```
Comment (ThomasMichon) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx]: The problem with it being a `<button>` is that it becomes focusable, when designers have specifically asked that it not be. The toggle-selection element for keyboarding is the row itself. The checkbox is a visual aid for mouse-targeting only.

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx (hunk 1 above)
Comment (HalcyonPride) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx]: Not exactly: https://www.w3.org/TR/2012/WD-html-markup-20121011/button.button.html

Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx]: It defaults to `type="submit"` -- https://www.w3.org/TR/html5/sec-forms.html#element-attrdef-button-type

But typically this only matters in the context of a form if you're trying to negate the submit behavior.

Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx]: Interesting... https://twitter.com/davidmacd/status/989170029961859073

Comment (HalcyonPride) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx]: So do we keep it or toss it? https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button confirms that it defaults to 'submit'

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx (hunk 2 above)
Comment (HalcyonPride) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx]: Any recommendations on how to fix the attached Bug w/o using `<button>`?

**Comment by HalcyonPride** [comment]
@ThomasMichon, could you please merge the PR for me?

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx (hunk 2 above)
Comment (HalcyonPride) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowCheck.tsx]: Also, as far as I know, any interactive element *should* be focus-able and Design should know this.

**Comment by HalcyonPride** [comment]
Design **does not** want the "checkbox" to be focus-able, so it shouldn't be a `<button>`. Furthermore, [this link](https://www.w3.org/TR/wai-aria-practices-1.1/examples/checkbox/checkbox-1/checkbox-1.html) confirms that the checkbox *should* work correctly as a `<div>` so this is a Narrator Bug instead.

**Pull request closed without merging**